packages/edit-site/src/store/selectors.js

```diff
@@ -25,3 +32,137 @@ export function isFeatureActive( state, feature ) {
 export function __experimentalGetPreviewDeviceType( state ) {
 	return state.deviceType;
 }
+
+/**
+ * Returns whether the current user can create media or not.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {Object} Whether the current user can create media or not.
+ */
+export const getCanUserCreateMedia = createRegistrySelector( ( select ) => () =>
+	select( 'core' ).canUser( 'create', 'media' )
+);
+
+/**
+ * Returns the settings, taking into account active features and permissions.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {Object} Settings.
+ */
+export const getSettings = createSelector(
+	( state ) => {
+		const canUserCreateMedia = getCanUserCreateMedia( state );
+		if ( ! canUserCreateMedia ) {
+			return state.settings;
+		}
+
+		return {
+			...state.settings,
+			focusMode: isFeatureActive( state, 'focusMode' ),
+			hasFixedToolbar: isFeatureActive( state, 'fixedToolbar' ),
+			mediaUpload( { onError, ...rest } ) {
+				uploadMedia( {
+					wpAllowedMimeTypes: state.settings.allowedMimeTypes,
+					onError: ( { message } ) => onError( message ),
+					...rest,
+				} );
+			},
+		};
+	},
+	( state ) => [
+		getCanUserCreateMedia( state ),
+		state.settings,
+		isFeatureActive( state, 'focusMode' ),
+		isFeatureActive( state, 'fixedToolbar' ),
+	]
+);
+
+/**
+ * Returns the current home template ID.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {number} Home template ID.
+ */
+export function getHomeTemplateId( state ) {
+	return state.homeTemplateId;
+}
+
+/**
+ * Returns the current template ID.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {number} Template ID.
+ */
+export function getTemplateId( state ) {
+	return state.templateId;
+}
+
+/**
+ * Returns the current template part ID.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {number} Template part ID.
+ */
+export function getTemplatePartId( state ) {
+	return state.templatePartId;
+}
+
+/**
+ * Returns the current template type.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {string} Template type.
+ */
+export function getTemplateType( state ) {
+	return state.templateType;
+}
+
+/**
+ * Returns the current template IDs.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {number[]} Template IDs.
+ */
+export function getTemplateIds( state ) {
+	return state.templateIds;
+}
```

Comment:
AFAICS, the `getTemplateIds` selector isn't used anywhere. We could consider dropping it (and the corresponding `templateIds` reducer, plus potentially some related actions, resolvers, and `editorInitialState`).

Reply:
I think they might come in handy down the road though?  If I'm understanding it right their values are being initialized from the `edit-site-page.php`, so now getting them will be easily accessible in future development.

Reply:
@ockham Worked on removing it from `edit-site` state and instead use the `getEntityRecords` API with a new query parameter.

I think that, in #21878, we should remove both the `templateIDs` and `templatePartIds` selectors and reducers, as well as the code in `edit-site-page` that loads it.